iocore/net/P_SNIActionPerformer.h

```diff
@@ -56,24 +45,26 @@ class ActionItem
   virtual ~ActionItem(){};
 };
 
-class DisableH2 : public ActionItem
+class ControlH2 : public ActionItem
 {
 public:
-  DisableH2() {}
-  ~DisableH2() override {}
+  ControlH2(YamlSNIConfig::Control control_value) : control(control_value) {}
+  ~ControlH2() override {}
 
   int
   SNIAction(Continuation *cont) const override
   {
-    auto ssl_vc     = dynamic_cast<SSLNetVConnection *>(cont);
-    auto accept_obj = ssl_vc ? ssl_vc->accept_object : nullptr;
-    if (accept_obj && accept_obj->snpa && ssl_vc) {
-      if (auto it = snpsMap.find(accept_obj->id); it != snpsMap.end()) {
-        ssl_vc->registerNextProtocolSet(it->second);
-      }
+    auto ssl_vc = dynamic_cast<SSLNetVConnection *>(cont);
+    if (control == YamlSNIConfig::Control::DISABLE) {
+      ssl_vc->disableProtocol(TS_ALPN_PROTOCOL_INDEX_HTTP_2_0);
```

Comment:
ssl_vc can be nullptr

Reply:
Not entirely sure that it can be.  I don't think these actions will be invoked outside of the SSL callbacks.  But better to be safe than sorry.  At least it will future proof the logic.